src/ImuSensor.cc

```diff
@@ -255,6 +256,43 @@ bool ImuSensor::Update(const std::chrono::steady_clock::duration &_now)
   frame->set_key("frame_id");
   frame->add_value(this->FrameId());
 
+  // Populate covariance
+  for (int i = 0; i < 9; ++i) {
+    msg.mutable_linear_acceleration_covariance()->add_data(0);
+    msg.mutable_angular_velocity_covariance()->add_data(0);
+    msg.mutable_orientation_covariance()->add_data(0);
+  }
+
+  auto getCov = [&](SensorNoiseType noiseType) -> float{
+    if (this->dataPtr->noises.find(noiseType) != this->dataPtr->noises.end()) {
+      GaussianNoiseModelPtr gaussian =
+        std::dynamic_pointer_cast<GaussianNoiseModel>(
+            this->dataPtr->noises[noiseType]);
+      if (gaussian) {
+        return static_cast<float>(gaussian->StdDev() * gaussian->StdDev());
+      }
+    }
+    return 0.001f;
```

Comment:
is `0.001f` a good default value to return? or `0`?

Reply:
It's good question. I'm not entirely sure here.
According to ros [imu message definition](http://docs.ros.org/en/noetic/api/sensor_msgs/html/msg/Imu.html):
- if the covariance is known it should be filled in
- if the covariance is unknown we should set it to 0
- if we're missing an estimate for one of those values, we should set it to -1.

In this case, I guess 0 is a good default, that is if we want to follow ros conventions in gz-sensors as well.

Reply:
ok let's keep it consistent with ROS and give it 0 here. Can you also do that for the `orientation_covariance` below?

Reply:
Friendly ping @ejalaa12, I would like to get this in for Harmonic.

Reply:
Hey sorry for the late reply.
I just updated the PR and set the default to 0.
Concerning the `orientation_covariance`, setting it 0. It is important to note that this implies (according ros convention) the noise is always unknown, while in fact, we are just not applying any. 
This is fine for robot_localization for example (a [small offset](http://docs.ros.org/en/noetic/api/robot_localization/html/preparing_sensor_data.html#:~:text=.-,If%20a,\)%20to%20that%20value.,-A%20better%20solution) will be added), but I wonder what this will do for other localization nodes... Thankfully, this will not break anything as the behavior would be unchanged with this PR :)


On a different tangent, should we consider applying noise to the orientation values (probably a separate PR). That would mean addind a new [SensorNoiseType](https://github.com/gazebosim/gz-sensors/blob/7b4387fc64b12f72a39e7c7efafcbc8cc4230766/include/gz/sensors/SensorTypes.hh#L103)

Reply:
Noising up orientation can be tricky as an IMU isn't really directly measuring that, but instead running it through some sort of filter.  You end up with things that aren't linear and uncorrelated in a tidy way.  I think we can open a ticket to follow up on that though, if you would like.